rtengine/diagonalcurves.cc

```diff
@@ -454,7 +447,13 @@ double DiagonalCurve::getVal (double t) const
             // add highlights effect:
             double htv = xexp(max(mhc * xlog((stretched - mfc) / (1.0 - mfc)),-236.0));
             double hbase = pfull_alt (htv, 0.5, x[4]);
-            return mfc + (1.0 - mfc) * xexp(xlog(hbase) / mhc);
+            //this part of the curve isn't affected by highlight, return the base curve
+            if (hbase < 1e-14 ){ 
+                hbase = xexp(max(xlog((stretched - mfc) / (1.0 - mfc)),-236.0));
+                return mfc + (1.0 - mfc) * hbase;
```

Comment:
If you write out the maths, this actually simplifies greatly to `return stretched`. Please check if you can.